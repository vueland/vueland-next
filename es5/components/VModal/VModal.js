"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VModal = void 0;

require("../../../src/components/VModal/VModal.scss");

var _vue = require("vue");

var _useOverlay = require("../../effects/use-overlay");

var _useTransition = require("../../effects/use-transition");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var modalProps = _objectSpread(_objectSpread({
  width: {
    type: [String, Number],
    "default": 400
  },
  show: Boolean
}, (0, _useOverlay.overlayProps)()), (0, _useTransition.transitionProps)());

var VModal = (0, _vue.defineComponent)({
  name: 'v-modal',
  props: modalProps,
  setup: function setup(props, _ref) {
    var slots = _ref.slots,
        emit = _ref.emit;

    if (props.overlay) {
      var overlay = (0, _useOverlay.useOverlay)(props, 'v-modal');
      (0, _vue.watch)(function () {
        return props.show;
      }, function (to) {
        to && overlay.createOverlay();
        !to && overlay.removeOverlay();
      });
    }

    var genContent = function genContent() {
      return (0, _vue.h)('div', {
        "class": {
          'v-modal__content': true
        }
      }, slots["default"] && slots["default"]());
    };

    var genModal = function genModal() {
      return (0, _vue.h)('div', {
        "class": {
          'v-modal': true
        },
        'onUpdate:show': function onUpdateShow(val) {
          return emit('update:show', val);
        }
      }, [content]);
    };

    var content = genContent();
    var modal = genModal();

    if (!!props.transition) {
      var createTransition = (0, _useTransition.useTransition)(props, modal);
      modal = createTransition();
    }

    return function () {
      return (0, _vue.withDirectives)((0, _vue.h)(modal), [[_vue.vShow, props.show]]);
    };
  }
});
exports.VModal = VModal;
//# sourceMappingURL=VModal.js.map