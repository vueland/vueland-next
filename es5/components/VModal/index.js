"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _VModal = require("./VModal");

Object.keys(_VModal).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _VModal[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _VModal[key];
    }
  });
});
//# sourceMappingURL=index.js.map