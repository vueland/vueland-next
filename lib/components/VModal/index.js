export * from "./VModal";
//# sourceMappingURL=index.js.map