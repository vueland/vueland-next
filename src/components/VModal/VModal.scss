@import "../../styles/scss/modules/global";

.v-modal {
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  padding: 20px;
  overflow: hidden;
}
