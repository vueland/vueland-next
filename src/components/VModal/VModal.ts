// Vue API
import {
  h,
  ref,
  watch,
  withDirectives,
  defineComponent,
  vShow,
  onMounted,
} from 'vue'

// Effects
import { overlayProps, useOverlay } from '../../composable/use-overlay'
import { transitionProps, useTransition } from '../../composable/use-transition'
import { useToggle } from '../../composable/use-toggle'

// Types
import { VNode } from 'vue'
import { convertToUnit } from '@/helpers'

export const VModal = defineComponent({
  name: 'v-modal',

  props: {
    modelValue: Boolean,
    width: {
      type: [Number, String],
      default: null,
    },
    ...overlayProps(),
    ...transitionProps(),
  } as any,

  emits: ['update:modelValue'],

  setup(props, { slots, emit }) {
    const { isActive } = useToggle(props)

    const modalRef = ref(null)

    onMounted(() => {
      if (props.overlay) {
        const { createOverlay, removeOverlay } = useOverlay(
          props,
          modalRef.value!,
        )

        isActive.value && createOverlay()

        watch(
          () => isActive.value,
          (to) => {
            to && createOverlay()
            !to && removeOverlay()
          },
        )
      }
    })

    function genContent(): VNode {
      const propsData = {
        class: 'v-modal__content',
        style: {
          width: props.width ? convertToUnit(props.width) : '',
        },
      }
      return h('div', propsData, slots.default && slots.default())
    }

    function genModal() {
      const propsData = {
        class: 'v-modal',
        ref: modalRef,
        ['onUpdate:modelValue']: (val) => emit('update:modelValue', val),
      }

      return withDirectives(h('div', propsData, genContent()), [
        [vShow, isActive.value],
      ])
    }

    return () => useTransition(genModal(), props.transition)
  },
})
