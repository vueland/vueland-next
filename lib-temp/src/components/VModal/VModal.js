import './VModal.scss';
import { h, ref, watch, withDirectives, defineComponent, vShow, onMounted, } from 'vue';
import { overlayProps, useOverlay } from '../../effects/use-overlay';
import { transitionProps, useTransition } from '../../effects/use-transition';
import { useToggle } from '../../effects/use-toggle';
export const VModal = defineComponent({
    name: 'v-modal',
    props: {
        modelValue: Boolean,
        ...overlayProps(),
        ...transitionProps(),
    },
    setup(props, { slots, emit }) {
        const { isActive } = useToggle(props);
        const modalRef = ref(null);
        onMounted(() => {
            if (props.overlay) {
                const { createOverlay, removeOverlay } = useOverlay(props, modalRef.value);
                isActive.value && createOverlay();
                watch(() => isActive.value, to => {
                    to && createOverlay();
                    !to && removeOverlay();
                });
            }
        });
        function genContent() {
            const propsData = {
                class: 'v-modal__content',
            };
            return h('div', propsData, slots.default && slots.default());
        }
        function genModal() {
            const propsData = {
                class: 'v-modal',
                ref: modalRef,
                ['onUpdate:modelValue']: val => emit('update:modelValue', val),
            };
            return withDirectives(h('div', propsData, genContent()), [[vShow, isActive.value]]);
        }
        return () => useTransition(genModal(), props.transition);
    },
});
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiVk1vZGFsLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiLi4vLi4vLi4vLi4vc3JjL2NvbXBvbmVudHMvVk1vZGFsL1ZNb2RhbC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFDQSxPQUFPLGVBQWUsQ0FBQTtBQUd0QixPQUFPLEVBQ0wsQ0FBQyxFQUNELEdBQUcsRUFDSCxLQUFLLEVBQ0wsY0FBYyxFQUNkLGVBQWUsRUFDZixLQUFLLEVBQ0wsU0FBUyxHQUNWLE1BQU0sS0FBSyxDQUFBO0FBR1osT0FBTyxFQUFFLFlBQVksRUFBRSxVQUFVLEVBQUUsTUFBTSwyQkFBMkIsQ0FBQTtBQUNwRSxPQUFPLEVBQUUsZUFBZSxFQUFFLGFBQWEsRUFBRSxNQUFNLDhCQUE4QixDQUFBO0FBQzdFLE9BQU8sRUFBRSxTQUFTLEVBQUUsTUFBTSwwQkFBMEIsQ0FBQTtBQUtwRCxNQUFNLENBQUMsTUFBTSxNQUFNLEdBQUcsZUFBZSxDQUFDO0lBQ3BDLElBQUksRUFBRSxTQUFTO0lBRWYsS0FBSyxFQUFFO1FBQ0wsVUFBVSxFQUFFLE9BQU87UUFDbkIsR0FBRyxZQUFZLEVBQUU7UUFDakIsR0FBRyxlQUFlLEVBQUU7S0FDZDtJQUVSLEtBQUssQ0FBQyxLQUFLLEVBQUUsRUFBRSxLQUFLLEVBQUUsSUFBSSxFQUFFO1FBQzFCLE1BQU0sRUFBRSxRQUFRLEVBQUUsR0FBRyxTQUFTLENBQUMsS0FBSyxDQUFDLENBQUE7UUFFckMsTUFBTSxRQUFRLEdBQUcsR0FBRyxDQUFDLElBQUksQ0FBQyxDQUFBO1FBRTFCLFNBQVMsQ0FBQyxHQUFHLEVBQUU7WUFDYixJQUFJLEtBQUssQ0FBQyxPQUFPLEVBQUU7Z0JBQ2pCLE1BQU0sRUFBRSxhQUFhLEVBQUUsYUFBYSxFQUFFLEdBQUcsVUFBVSxDQUNqRCxLQUFLLEVBQ0wsUUFBUSxDQUFDLEtBQU0sQ0FDaEIsQ0FBQTtnQkFFRCxRQUFRLENBQUMsS0FBSyxJQUFJLGFBQWEsRUFBRSxDQUFBO2dCQUVqQyxLQUFLLENBQ0gsR0FBRyxFQUFFLENBQUMsUUFBUSxDQUFDLEtBQUssRUFDcEIsRUFBRSxDQUFDLEVBQUU7b0JBQ0gsRUFBRSxJQUFJLGFBQWEsRUFBRSxDQUFBO29CQUNyQixDQUFDLEVBQUUsSUFBSSxhQUFhLEVBQUUsQ0FBQTtnQkFDeEIsQ0FBQyxDQUNGLENBQUE7YUFDRjtRQUNILENBQUMsQ0FBQyxDQUFBO1FBRUYsU0FBUyxVQUFVO1lBQ2pCLE1BQU0sU0FBUyxHQUFHO2dCQUNoQixLQUFLLEVBQUUsa0JBQWtCO2FBQzFCLENBQUE7WUFDRCxPQUFPLENBQUMsQ0FBQyxLQUFLLEVBQUUsU0FBUyxFQUFFLEtBQUssQ0FBQyxPQUFPLElBQUksS0FBSyxDQUFDLE9BQU8sRUFBRSxDQUFDLENBQUE7UUFDOUQsQ0FBQztRQUVELFNBQVMsUUFBUTtZQUNmLE1BQU0sU0FBUyxHQUFHO2dCQUNoQixLQUFLLEVBQUUsU0FBUztnQkFDaEIsR0FBRyxFQUFFLFFBQVE7Z0JBQ2IsQ0FBQyxxQkFBcUIsQ0FBQyxFQUFFLEdBQUcsQ0FBQyxFQUFFLENBQUMsSUFBSSxDQUFDLG1CQUFtQixFQUFFLEdBQUcsQ0FBQzthQUMvRCxDQUFBO1lBRUQsT0FBTyxjQUFjLENBQ25CLENBQUMsQ0FBQyxLQUFLLEVBQUUsU0FBUyxFQUFFLFVBQVUsRUFBRSxDQUFDLEVBQ2pDLENBQUMsQ0FBQyxLQUFLLEVBQUUsUUFBUSxDQUFDLEtBQUssQ0FBQyxDQUFDLENBQzFCLENBQUE7UUFDSCxDQUFDO1FBRUQsT0FBTyxHQUFHLEVBQUUsQ0FBQyxhQUFhLENBQUMsUUFBUSxFQUFFLEVBQUUsS0FBSyxDQUFDLFVBQVUsQ0FBQyxDQUFBO0lBQzFELENBQUM7Q0FDRixDQUFDLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyIvLyBTdHlsZXNcbmltcG9ydCAnLi9WTW9kYWwuc2NzcydcblxuLy8gVnVlIEFQSVxuaW1wb3J0IHtcbiAgaCxcbiAgcmVmLFxuICB3YXRjaCxcbiAgd2l0aERpcmVjdGl2ZXMsXG4gIGRlZmluZUNvbXBvbmVudCxcbiAgdlNob3csXG4gIG9uTW91bnRlZCxcbn0gZnJvbSAndnVlJ1xuXG4vLyBFZmZlY3RzXG5pbXBvcnQgeyBvdmVybGF5UHJvcHMsIHVzZU92ZXJsYXkgfSBmcm9tICcuLi8uLi9lZmZlY3RzL3VzZS1vdmVybGF5J1xuaW1wb3J0IHsgdHJhbnNpdGlvblByb3BzLCB1c2VUcmFuc2l0aW9uIH0gZnJvbSAnLi4vLi4vZWZmZWN0cy91c2UtdHJhbnNpdGlvbidcbmltcG9ydCB7IHVzZVRvZ2dsZSB9IGZyb20gJy4uLy4uL2VmZmVjdHMvdXNlLXRvZ2dsZSdcblxuLy8gVHlwZXNcbmltcG9ydCB7IFZOb2RlIH0gZnJvbSAndnVlJ1xuXG5leHBvcnQgY29uc3QgVk1vZGFsID0gZGVmaW5lQ29tcG9uZW50KHtcbiAgbmFtZTogJ3YtbW9kYWwnLFxuXG4gIHByb3BzOiB7XG4gICAgbW9kZWxWYWx1ZTogQm9vbGVhbixcbiAgICAuLi5vdmVybGF5UHJvcHMoKSxcbiAgICAuLi50cmFuc2l0aW9uUHJvcHMoKSxcbiAgfSBhcyBhbnksXG5cbiAgc2V0dXAocHJvcHMsIHsgc2xvdHMsIGVtaXQgfSkge1xuICAgIGNvbnN0IHsgaXNBY3RpdmUgfSA9IHVzZVRvZ2dsZShwcm9wcylcblxuICAgIGNvbnN0IG1vZGFsUmVmID0gcmVmKG51bGwpXG5cbiAgICBvbk1vdW50ZWQoKCkgPT4ge1xuICAgICAgaWYgKHByb3BzLm92ZXJsYXkpIHtcbiAgICAgICAgY29uc3QgeyBjcmVhdGVPdmVybGF5LCByZW1vdmVPdmVybGF5IH0gPSB1c2VPdmVybGF5KFxuICAgICAgICAgIHByb3BzLFxuICAgICAgICAgIG1vZGFsUmVmLnZhbHVlISxcbiAgICAgICAgKVxuXG4gICAgICAgIGlzQWN0aXZlLnZhbHVlICYmIGNyZWF0ZU92ZXJsYXkoKVxuXG4gICAgICAgIHdhdGNoKFxuICAgICAgICAgICgpID0+IGlzQWN0aXZlLnZhbHVlLFxuICAgICAgICAgIHRvID0+IHtcbiAgICAgICAgICAgIHRvICYmIGNyZWF0ZU92ZXJsYXkoKVxuICAgICAgICAgICAgIXRvICYmIHJlbW92ZU92ZXJsYXkoKVxuICAgICAgICAgIH0sXG4gICAgICAgIClcbiAgICAgIH1cbiAgICB9KVxuXG4gICAgZnVuY3Rpb24gZ2VuQ29udGVudCgpOiBWTm9kZSB7XG4gICAgICBjb25zdCBwcm9wc0RhdGEgPSB7XG4gICAgICAgIGNsYXNzOiAndi1tb2RhbF9fY29udGVudCcsXG4gICAgICB9XG4gICAgICByZXR1cm4gaCgnZGl2JywgcHJvcHNEYXRhLCBzbG90cy5kZWZhdWx0ICYmIHNsb3RzLmRlZmF1bHQoKSlcbiAgICB9XG5cbiAgICBmdW5jdGlvbiBnZW5Nb2RhbCgpIHtcbiAgICAgIGNvbnN0IHByb3BzRGF0YSA9IHtcbiAgICAgICAgY2xhc3M6ICd2LW1vZGFsJyxcbiAgICAgICAgcmVmOiBtb2RhbFJlZixcbiAgICAgICAgWydvblVwZGF0ZTptb2RlbFZhbHVlJ106IHZhbCA9PiBlbWl0KCd1cGRhdGU6bW9kZWxWYWx1ZScsIHZhbCksXG4gICAgICB9XG5cbiAgICAgIHJldHVybiB3aXRoRGlyZWN0aXZlcyhcbiAgICAgICAgaCgnZGl2JywgcHJvcHNEYXRhLCBnZW5Db250ZW50KCkpLFxuICAgICAgICBbW3ZTaG93LCBpc0FjdGl2ZS52YWx1ZV1dLFxuICAgICAgKVxuICAgIH1cblxuICAgIHJldHVybiAoKSA9PiB1c2VUcmFuc2l0aW9uKGdlbk1vZGFsKCksIHByb3BzLnRyYW5zaXRpb24pXG4gIH0sXG59KVxuIl19