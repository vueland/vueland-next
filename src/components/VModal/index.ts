import VModal from './VModal'

export { VModal }
export default VModal
