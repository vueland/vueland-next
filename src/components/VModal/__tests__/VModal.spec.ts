import { mount, VueWrapper } from '@vue/test-utils'
import { VModal } from '../VModal'
import 'regenerator-runtime/runtime'

describe('VModal', () => {
  let mountFunction: (options?: object) => VueWrapper<any>

  beforeEach(() => {
    mountFunction = (options = {}) => mount(VModal, { ...options })
  })

  it('should mount component and match snapshot', () => {
    const cmp = mountFunction()
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set overlay and match snapshot', () => {
    const overlay = true
    const props = { overlay }
    const cmp = mountFunction({ props })

    expect(cmp.props().overlay).toBe(true)
    expect(cmp.find('.v-overlay').exists()).toBe(true)
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set overlay color and match snapshot', () => {
    const overlayColor = 'red darken-2'
    const props = { overlayColor, overlay: true }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-overlay').attributes().class).toContain(overlayColor)
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set transition and match snapshot', () => {
    const transition = 'fade'
    const props = { transition }
    const cmp = mountFunction({ props })

    expect(cmp.props().transition).toBe(transition)
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set modelValue prop to true and match snapshot', async () => {
    const modelValue = true
    const props = { modelValue, overlay: true }
    const cmp = mountFunction({ props })

    // Waiting for next tick
    await new Promise(res => setTimeout(res))

    expect(cmp.find('.v-overlay').attributes().class).toContain('v-overlay--active')
    expect(cmp.html()).toMatchSnapshot()
  })
})
