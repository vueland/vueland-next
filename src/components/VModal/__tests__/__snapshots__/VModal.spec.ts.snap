// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VModal should mount component and match snapshot 1`] = `
<transition-stub>
  <div class="v-modal"
       show="false"
       style="display: none;"
  >
  </div>
</transition-stub>
`;

exports[`VModal should set modelValue prop to true and match snapshot 1`] = `
<transition-stub>
  <div class="v-overlay v-overlay--active"
       style="background-color: rgb(0, 0, 0); border-color: #000000;"
  >
  </div>
</transition-stub>
<transition-stub>
  <div class="v-modal"
       show="true"
  >
  </div>
</transition-stub>
`;

exports[`VModal should set overlay and match snapshot 1`] = `
<transition-stub>
  <div class="v-overlay v-overlay--hidden"
       style="background-color: rgb(0, 0, 0); border-color: #000000; display: none;"
  >
  </div>
</transition-stub>
<transition-stub>
  <div class="v-modal"
       show="false"
       style="display: none;"
  >
  </div>
</transition-stub>
`;

exports[`VModal should set overlay color and match snapshot 1`] = `
<transition-stub>
  <div class="v-overlay v-overlay--hidden red darken-2"
       style="display: none;"
  >
  </div>
</transition-stub>
<transition-stub>
  <div class="v-modal"
       show="false"
       style="display: none;"
  >
  </div>
</transition-stub>
`;

exports[`VModal should set transition and match snapshot 1`] = `
<transition-stub>
  <div class="v-modal"
       show="false"
       style="display: none;"
  >
  </div>
</transition-stub>
`;
